exporter/splunkhecexporter/client.go

```diff
@@ -46,14 +46,15 @@ const minCompressionLen = 1500
 
 // client sends the data to the splunk backend.
 type client struct {
-	config         *Config
-	url            *url.URL
-	healthCheckURL *url.URL
-	client         *http.Client
-	logger         *zap.Logger
-	wg             sync.WaitGroup
-	headers        map[string]string
-	gzipWriterPool *sync.Pool
+	config            *Config
+	url               *url.URL
+	healthCheckURL    *url.URL
+	client            *http.Client
+	logger            *zap.Logger
+	wg                sync.WaitGroup
+	headers           map[string]string
+	telemetrySettings component.TelemetrySettings
+	gzipWriterPool    *sync.Pool
```

Comment:
You shouldn't need this as far as I understand since it will be part of the configured transport for the http.Client.

Keep both may result in double compression?

Reply:
No, we manage compression ourselves here -we have a whole thing going on to batch into a gzipped payload.
You can see it in the setup, we explicitly disable compression if set via HTTPClientSettings.
Maybe we could disambiguate more, not sure.

Reply:
I am referring to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16839/files#diff-db010afef1a221371796af2d113eb220bc280c6bc349e9e10cfe61e386f39461R894